authoringAssistant/src/main/java/explorableviz/transparenttext/TestQueryContext.java

```diff
@@ -12,105 +11,67 @@
-        JSONObject testCase = new JSONObject(content);
+    public static TestQueryContext importFromJson(String filePath) throws IOException {
+        JSONObject testCase = new JSONObject(new String(Files.readAllBytes(Path.of(STR."\{filePath}.json"))));
         JSONArray json_datasets = testCase.getJSONArray("datasets");
         JSONObject json_variables = testCase.getJSONObject("variables");
         JSONArray json_imports = testCase.getJSONArray("imports");
-        String code = new String(Files.readAllBytes(Path.of(STR."\{filePath}.fld")));
         JSONArray json_paragraph = testCase.getJSONArray("paragraph");
-        ArrayList<TextFragment> paragraph = new ArrayList<>();
 
-        for (int i = 0; i < json_paragraph.length(); i++) {
-            JSONObject paragraph_element = json_paragraph.getJSONObject(i);
-            String type = paragraph_element.getString("type");
-            switch (type) {
-                case "literal":
-                    paragraph.add(new Literal(paragraph_element.getString("value")));
-                    break;
-                case "expression":
-                    paragraph.add(new Expression(json_paragraph.getJSONObject(i).getString("expression"), json_paragraph.getJSONObject(i).getString("value")));
-                    break;
-                default:
-                    throw new RuntimeException((STR."\{paragraph_element.getString("type")} type is invalid"));
-            }
-        }
-        String expected = testCase.getString("expected");
+        List<TextFragment> paragraph = IntStream.range(0, json_paragraph.length())
+                .mapToObj(i -> {
+                    JSONObject paragraphElement = json_paragraph.getJSONObject(i);
+                    String type = paragraphElement.getString("type");
 
-        HashMap<String, String> variables = new HashMap<>();
-        HashMap<String, String> datasets = new HashMap<>();
-        ArrayList<String> imports = new ArrayList<>();
+                    return switch (type) {
+                        case "literal" -> new Literal(paragraphElement.getString("value"));
+                        case "expression" ->
+                                new Expression(paragraphElement.getString("expression"), paragraphElement.getString("value"));
+                        default -> throw new RuntimeException(STR."\{type} type is invalid");
+                    };
+                })
+                .collect(Collectors.toList());
 
-        for (String key : json_variables.keySet()) {
-            variables.put(key, json_variables.getString(key));
-        }
-        for (int i = 0; i < json_datasets.length(); i++) {
-            datasets.put(json_datasets.getJSONObject(i).getString("var"), json_datasets.getJSONObject(i).getString("file"));
-        }
-        for (int i = 0; i < json_imports.length(); i++) {
-            imports.add(json_imports.getString(i));
-        }
+        Map<String, String> variables = json_variables.keySet().stream()
```

Comment:
Glad you made this change, I was going to suggest this yesterday as well (but didn’t want to overwhelm!)